src/main/java/frc/robot/utilities/LinearRegression.java

```diff
@@ -0,0 +1,141 @@
+package frc.robot.utilities;
+
+/**
+ *  The {@code LinearRegression.java} class performs a simple linear regression
+ *  on an set of <em>n</em> data points (<em>y<sub>i</sub></em>, <em>x<sub>i</sub></em>).
+ *  That is, it fits a straight line <em>y</em> = &alpha; + &beta; <em>x</em>,
+ *  (where <em>y</em> is the response variable, <em>x</em> is the predictor variable,
+ *  &alpha; is the <em>y-intercept</em>, and &beta; is the <em>slope</em>)
+ *  that minimizes the sum of squared residuals of the linear regression model.
+ *  It also computes associated statistics, including the coefficient of
+ *  determination <em>R</em><sup>2</sup> and the standard deviation of the
+ *  estimates for the slope and <em>y</em>-intercept.
+ *
+ */
+
+public class LinearRegression {
```

Comment:
Either add this code as a library or if this doesn't exist as a library state the source of the code with {@link link}